src/helpers/glpTEST/getGlpSwapData.js

```diff
@@ -0,0 +1,286 @@
+    return {
+      address: token,
+      maxAmountIn: maxAmountInArr[idx].gt(testLimit)
+        ? testLimit
+        : maxAmountInArr[idx],
+    };
+  });
+
+  return tokensInfo;
+};
+
+const getGlpLevData = async (
+  cookData,
+  provider,
+  pool,
+  sellAmount,
+  chainId,
+  slipage
+) => {
+  // store.commit("updateRouteData", []);
+  // store.commit("setPopupState", {
+  //   type: "test",
+  //   isShow: true,
+  // });
+
+  const { borrowToken, collateralToken } = pool;
+
+  const tokensArr = await getWhitelistedTokens();
+
+  // 0x to many request
+  // const initialRequestsArr = await axios.all(
+  //   tokensArr.map((token) =>
+  //     swap0xRequest(
+  //       chainId,
+  //       token.address,
+  //       borrowToken.address,
+  //       slipage,
+  //       sellAmount
+  //     )
+  //   )
+  // );
+
+  const initialRequestsArr = [];
+
+  for (let token of tokensArr) {
+    initialRequestsArr.push(
+      await swap0xRequest(
+        chainId,
+        token.address,
+        borrowToken.address,
+        slipage,
+        sellAmount
+      )
+    );
+  }
+
+  const mintedGlpFromTokenInArr = await Promise.all(
+    initialRequestsArr.map((resp) =>
+      gmxLensContract.getMintedGlpFromTokenIn(resp.buyToken, resp.buyAmount)
+    )
+  );
+
+  const initialResults = initialRequestsArr.map((resp, idx) => {
+    return {
+      ...resp,
+      ...mintedGlpFromTokenInArr[idx],
+    };
+  });
+
+  const tokenInfoArray = initialResults.sort(maxGlpAmount);
+
+  store.commit("updateRouteData", tokenInfoArray);
+
+  let globalLimit = ethers.BigNumber.from(0);
+  tokenInfoArray.forEach((item) => {
+    globalLimit = globalLimit.add(item.maxAmountIn);
+  });
+
+  if (sellAmount.gt(globalLimit)) console.log("to much mim to swap"); // TODO: add notification
+
+  let mimLeftToSwap = sellAmount;
+  const cookInfo = [];
+
+  for (let info of tokenInfoArray) {
+    if (mimLeftToSwap.eq(0)) break;
+
+    const itsAmountEnough =
+      info.maxAmountIn.gt(mimLeftToSwap) || info.maxAmountIn.eq(mimLeftToSwap);
+    const awailableToSwap = itsAmountEnough ? mimLeftToSwap : info.maxAmountIn;
+
+    if (itsAmountEnough && cookInfo.length === 0) {
+      cookInfo.push(info);
+    } else {
+      const { data, buyAmount, sellAmount } = await swap0xRequest(
+        chainId,
+        info.buyToken,
+        borrowToken.address,
+        slipage,
+        awailableToSwap
+      );
+
+      info.data = data;
+      info.buyAmount = buyAmount;
+      info.sellAmount = sellAmount;
+
+      cookInfo.push(info);
+    }
+
+    mimLeftToSwap = mimLeftToSwap.sub(awailableToSwap);
+  }
+
+  const mintedGlpFromTokenInArrFinal = await Promise.all(
+    cookInfo.map((info) =>
+      gmxLensContract.getMintedGlpFromTokenIn(info.buyToken, info.buyAmount)
+    )
+  );
+  const minExpectedArr = await Promise.all(
+    mintedGlpFromTokenInArrFinal.map((mintedGlpFromTokenIn) =>
+      collateralToken.contract.convertToShares(mintedGlpFromTokenIn.amountOut)
```

Comment:
should we add slippage to this minExpected?

Reply:
it is calculated from buyAmount and already takes this into account as far as I understand it, this is not correct?

Reply:
buyAmount does not take slippage into account, you need to fix this for both lev and liq

Reply:
buyAmount / sellAmount in the 0x response do not include slippage, maybe you could adapt swap0xRequest to return also buyAmountWithSlippage and sellAmountWithSlippage and consume those accordingly @0xhardrefresh

Reply:
@0xm3rlin  updated with enableSlippageProtection, what you think about this?